hack/validate/shfmt

```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env bash
 set -e -o pipefail
 
-shfmtflags=(-bn -ci -sr)
+shfmtflags=(-bn -ci -sr -d "$@") # Allows to specify additional flags.
 # NOTE: `git grep '^#!'` may also pick up non-shell script files.
 # Add exceptional files to `grep -v` if any false-positive is detected.
 if git grep --name-only '^#!' | grep -E -v '(vendor|\.go|Jenkinsfile|\.py|\.bats)' \
-	| xargs shfmt -d "${shfmtflags[@]}"; then
+	| xargs shfmt "${shfmtflags[@]}"; then
 	echo 'Congratulations! The shell scripts are properly formatted.'
 else
-	echo "Please reformat the shell scripts with \`shfmt -w ${shfmtflags[*]}\`."
+	echo "Please reformat the shell scripts with \`$0 -w\`."
```

Comment:
Do we have a (e.g.) `make shfmt` target? I'd still be ok with it only _printing_ instructions, as long as we have an easy-to-consume `make` target (with all the right options set).

Reply:
I looked and we don't. If we want one, it's easy to add, except for one thing -- this script can be used with or without `-w`, and making a similar thing with the make target is slightly more complicated.

Reply:
Basically, if we are to add it as a make target, we want it to run with `-w` (i.e. actually change the files), and the whole validation process would be slightly different. Currently, we have a shell script which runs (or sources -- before this PR) a bunch of other shell scripts, and adding a call to make will slightly ripple the uniformity of that approach.